containers.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2034,SC2143
+set -euo pipefail
+
+# Network array - to add more, separate by space and double qoutes e.g. "todo-app" "your-second-network"
+networks=("todo-app")
+# Volume array - to add more, separate by space and double qoutes, e.g. "todo-mysql-data" "your-second-volume"
+volumes=("todo-mysql-data")
+# Path to docker images
+images="${PWD}/docker/images"
+# Set app container name
+app="todo-app"
+# Set db container name
+db="todo-db"
+
+log(){
+  echo "$(date +'%T') $1"
+}
+
+networks(){
+  docker network create \
+    "${network}"
+}
+
+volumes(){
+  docker volume create \
+    "${volume}"
+}
+
+todo_app(){
+  docker run \
+    --detach \
+    --name "${app}" \
+    --publish 3000:3000 \
+    --workdir "/app" \
+    --volume "${images}/getting-started/app:/app" \
+    --network todo-app \
+    --env MYSQL_HOST=mysql \
+    --env MYSQL_USER=root \
+    --env MYSQL_PASSWORD=secret \
+    --env MYSQL_DB=todos \
+    node:18-alpine \
+    sh -c "yarn install && yarn run dev"
+}
+
+todo_db(){
+  docker run \
+    --detach \
+    --name "${db}" \
+    --network todo-app \
+    --network-alias mysql \
+    --volume todo-mysql-data:/var/lib/mysql \
+    --env MYSQL_ROOT_PASSWORD=secret \
+    --env MYSQL_DATABASE=todos \
+     mysql:5.7
+}
+
+# Create networks for containers
+log "Checking for existing networks..."
+for network in "${networks[@]}"
+do
+  if [ "$(docker network ls | grep "${network}")" ]; then
```

Comment:
It will check both entries individually and only create networks not currently existing. I think I know what you are getting at, but I've tested my grep functionality extensively and it does what I expect.

Reply:
First iteration ("todo-app"): A network matching "todo-app" does not exist, so it is created
Second iteration: A network matching "app" _does_ exist ("todo-app" contains "app"), so no "app" network is created.